utils/emmylua/wesnoth.lua

```diff
@@ -0,0 +1,267 @@
+---@meta
+
+wesnoth = {}
+
+---@return "'Game Lua Kernel'" | "'Mapgen Lua Kernel'" | "'Application Lua Kernel'"
+function wesnoth.kernel_type() end
+
+---@param path string
+---@return ...
+function wesnoth.dofile(path, ...) end
+
+---@class tstring : string
+
+---Constructs a textdomain, which can be called to create translatable strings.
+---@param domain string The textdomain name
+---@return fun(str:string, str_plural?:string, n?:number):tstring
+function wesnoth.textdomain(domain) end
+
+---Logs a message to the console
+---@param logger "'info'"|"'debug'"|"'warning'"|"'error'"|"'wml'"
+---@param message string
+---@param in_chat boolean
+---@overload fun(message:string, in_chat:boolean)
+function wesnoth.log(logger, message, in_chat) end
+
+---Simulate combat between two units_MP
+---@param attacker unit
+---@param attacker_weapon integer
+---@param defender unit
+---@param defender_weapon integer
+---@return stats_evaluation attacker_stats
+---@return stats_evaluation defender_stats
+---@return weapon_evaluation attacker_weapon
+---@return weapon_evaluation defender_weapon
+function wesnoth.simulate_combat(attacker, attacker_weapon, defender, defender_weapon) end
+
+---@class stats_evaluation
+---@field poisoned number
+---@field slowed number
+---@field untouched number
+---@field average_hp number
+---@field hp_chance number[]
```

Comment:
This may be correct, if I'm misunderstanding the syntax. Does this say it's an array, or a table that's acting as a sparse array?

IIUC, it uses possible HP numbers as indicies, and the probability of that exact HP as values.

Reply:
I remember griping about this awhile back, which means `table<number, number>` would probably be more correct here.

Reply:
Should that be `table<integer, number>`?

Reply:
For some reason I thought the docs parser wouldn't accept that, but it seems like I was wrong.